.github/workflows/build-pr.yml

```diff
@@ -3,14 +3,17 @@ on:
   pull_request:
     branches:
       - main
+  push:
+    branches:
+      - main
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source code
-        uses: actions/checkout@v2
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # pin@v3.2.0
       - name: Setup Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # pin@3.5.0
         with:
           go-version: '1.18'
```

Comment:
This is more of a question/suggestion than a "we should do this". Curious if there would be any value in using the `go-version-file` input with the setup-go action. It takes the path to the `go.mod` file, and uses the version specified there, so when the go version is updated, it would automatically be reflected in the workflows. [Link](https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file) to the docs. My thinking is that it would be one less place to worry about updates/maintenance, but I'm curious to hear others' thoughts.

Reply:
thats a new(er) enhancement and I like that it will reduce the number of changes that need to be made when adopting new versions of go.

Thanks for calling it out!

Reply:
@lucasrod16 I didn't know about this feature. Very nice!

Reply:
@lucasrod16 for now, we have to revert back to specifying the version explicitly. Because it appears there was a regression (mismatch between what's actually in go.mod and the circle tests), that was introduced here 

https://github.com/helm/helm/blob/017785a2f1104eb4299f8d31feb550c213350f13/go.mod#L3

in this commit https://github.com/helm/helm/pull/10912/commits/ffa19a4b5d836283a91a4c16f8b81e734a973afc (see line https://github.com/helm/helm/pull/10912/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6) from this PR #10912. Note the commit comment:

> fix circleci tests

We revert go.mod back to 1.18 here but that would be scope creep for this PR. Let's do that in a follow-up PR, at which point we could use the `go-version-file` option.

Reply:
Made follow-up issue https://github.com/helm/helm/issues/11838

Reply:
@scottrigby ahh I see, thanks for the context! That sounds good, I can write-up an issue and link this discussion for more context.

@sabre1041 apologies for the side-track here

Reply:
@scottrigby nevermind I see that you've created a follow-up [issue](https://github.com/helm/helm/issues/11838) already.